tests/geometry/test_transformations.py

```diff
@@ -239,7 +239,7 @@ class VectorToRasterTestCase:
             no_data_value=0,
             raster_dtype=bool,
         ),
-        img_exp_statistics={"exp_shape": (100, 150, 1), "exp_dtype": bool, "exp_min": False, "exp_max": True},
+        img_exp_statistics={"exp_shape": (100, 150, 1), "exp_dtype": bool, "exp_min": np.False_, "exp_max": np.True_},
```

Comment:
I hate this change, but the alternative is doing some casting in sh-py ... Or we limit it to numpy < 2? but this would negatively affect field boundaries..

Reply:
I don't think we want to add such a harsh restriction. The reintroduced numpy booleans are causing some headaches...

Reply:
Do you suggest to update `sh-py`, then? It would be cleaner, but we would need to release sh-py afterwards

Reply:
what would you change there?

Reply:
The issue is that this tests uses the `assert_statistics_match` function, which lives in `sentinelhub/testing_utils.py`, so when the statistics are computed, they have the `numpy` dtypes

Reply:
but is that not correct? does the image not use numpy dtypes?

Reply:
Discussed in private to leave it as it is, the main issue in fact stems from the `pytest.approx` comparison, where the comparison doesn't yet work for the new numpy boolean types, meaning we need to have the exact types. 

![image](https://github.com/user-attachments/assets/0e3b8055-efc9-4b6e-beec-b219b70163f0)

![image](https://github.com/user-attachments/assets/abc5925b-9758-41dd-b80c-c29ecc79d710)

Perhaps this will be updated in the future. If that happens, we can simply revert the change.